docs/v4/concepts/di.md

```diff
@@ -76,3 +78,35 @@ $app->get('/foo', function (Request $request, Response $response, $args) {
     return $response;
 });
 ```
+
+## Configuring the application via a container
+
+If you have pre-configured container (e.g. for tests), you can call 
+the `AppFactory::createFromContainer()` method to create an `App`.
+
+**Example**
+
+```php
+<?php
+
+use DI\Container;
+use Slim\Factory\AppFactory;
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+// Create Container using PHP-DI
+$container = new Container();
+
+// Add a service to the container
+$container->set(MyService::class, function () {
```

Comment:
I think the example service should be one that is required by the `App` constructor, such as `ResponseFactoryInterface`. That's what that helper is for, in case you want to create the `App` using dependencies already defined in your container.

```
// Add a service to the container
$container->set(ResponseFactoryInterface::class, function () {
    return new Slim\Psr7\Factory\ResponseFactory();
});
```

Reply:
I will create a new PR (because my old repo has been deleted in the meantime).